core/src/main/java/hudson/model/View.java

```diff
@@ -1186,12 +1186,16 @@ public void updateByXml(Source source) throws IOException {
         }
 
         // try to reflect the changes by reloading
-
         try (InputStream in = new BufferedInputStream(new ByteArrayInputStream(out.toString().getBytes("UTF-8")))){
             // Do not allow overwriting view name as it might collide with another
             // view in same ViewGroup and might not satisfy Jenkins.checkGoodName.
             String oldname = name;
-            Jenkins.XSTREAM.unmarshal(new Xpp3Driver().createReader(in), this);
+            Object o = Jenkins.XSTREAM.unmarshal(new Xpp3Driver().createReader(in), this);
+            if (o!=this) {
```

Comment:
I was basing myself on the "tried, tested and true":

https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/model/AbstractItem.java#L673